models/zeroex/optimism/zeroex_api_optimism_fills.sql

```diff
@@ -0,0 +1,365 @@
+{{  config(
+        alias='optimism_fills',
+        materialized='incremental',
+        partition_by = ['block_date'],
+        unique_key = ['block_date', 'tx_hash', 'evt_index'],
+        on_schema_change='sync_all_columns',
+        file_format ='delta',
+        incremental_strategy='merge',
+        post_hook='{{ expose_spells(\'["optimism"]\',
+                                "project",
+                                "zeroex",
+                                \'["rantumBits", "sui414", "bakabhai993"]\') }}'
+    )
+}}
+
+
+
+-- Test Query here: https://dune.com/queries/1685501
+
+WITH zeroex_tx AS (
+    SELECT tx_hash,
+           max(affiliate_address) as affiliate_address
+    FROM (
+        SELECT tr.tx_hash,
+                    '0x' || CASE
+                                WHEN POSITION('869584cd' IN INPUT) <> 0
+                                THEN SUBSTRING(INPUT
+                                        FROM (position('869584cd' IN INPUT) + 32)
+                                        FOR 40)
+                                WHEN POSITION('fbc019a7' IN INPUT) <> 0
+                                THEN SUBSTRING(INPUT
+                                        FROM (position('fbc019a7' IN INPUT) + 32)
+                                        FOR 40)
+                            END AS affiliate_address
+        FROM {{ source('optimism', 'traces') }} tr
+        WHERE tr.to IN (
+                -- exchange contract
+                '0x61935cbdd02287b511119ddb11aeb42f1593b7ef', 
+                -- forwarder addresses
+                '0x6958f5e95332d93d21af0d7b9ca85b8212fee0a5',
+                '0x4aa817c6f383c8e8ae77301d18ce48efb16fd2be',
+                '0x4ef40d1bf0983899892946830abf99eca2dbc5ce', 
+                -- exchange proxy
+                '0xdef1abe32c034e558cdd535791643c58a13acc10'
+                )
+                AND (
+                        POSITION('869584cd' IN INPUT) <> 0
+                        OR POSITION('fbc019a7' IN INPUT) <> 0
+                    )
+                
+                {% if is_incremental() %}
+                AND block_time >= date_trunc('day', now() - interval '1 week') 
+                {% endif %}
+                {% if not is_incremental() %}
+             --   AND block_time >= '{{zeroex_v3_start_date}}'
+                {% endif %}
```

Comment:
Though the code logics is legitimate, we still need a default incremental filter for this part, as optimism.traces is a large table.
```suggestion
        FROM {{ source('optimism', 'traces') }} tr
        WHERE tr.to IN (
                -- exchange contract
                '0x61935cbdd02287b511119ddb11aeb42f1593b7ef', 
                -- forwarder addresses
                '0x6958f5e95332d93d21af0d7b9ca85b8212fee0a5',
                '0x4aa817c6f383c8e8ae77301d18ce48efb16fd2be',
                '0x4ef40d1bf0983899892946830abf99eca2dbc5ce', 
                -- exchange proxy
                '0xdef1abe32c034e558cdd535791643c58a13acc10'
                )
                AND (
                        POSITION('869584cd' IN INPUT) <> 0
                        OR POSITION('fbc019a7' IN INPUT) <> 0
                    )
                
                {% if is_incremental() %}
                AND block_time >= date_trunc('day', now() - interval '1 week') 
                {% endif %}
                {% if not is_incremental() %}
                AND block_time >= '{{zeroex_v3_start_date}}'
                {% endif %}
```